### DISCUSSION TITLE
Client usage

### DISCUSSION THREAD

**Discussion by etienne-sf** [enisdenjo/graphql-ws#249] [Q&A] [answered]
As stated in #247, I'm working on a java implementation for the ``

I've tried to use run a client.js script, that contains the code available [in your client snippet](https://github.com/enisdenjo/graphql-ws#use-the-client).
This client.js file is in the root of my node server project, configured according (I think) what describe in your readme page.

When I start it with the `node client.js` command, I get this error:

`Error: WebSocket implementation missing`

How can I solve this ?

Etienne

**Comment by enisdenjo** [comment, marked as answer]
Node does not provide you with a WebSocket implementation in the global, you have to supply one manually. Please consult the ["Client usage in Node" recipe](https://github.com/enisdenjo/graphql-ws#node-client).
